consensus/bor/api.go

```diff
@@ -75,13 +75,37 @@ func rankMapDifficulties(values map[common.Address]uint64) []difficultiesKV {
 }
 
 // GetSnapshotProposerSequence retrieves the in-turn signers of all sprints in a span
-func (api *API) GetSnapshotProposerSequence(number *rpc.BlockNumber) (BlockSigners, error) {
-	snapNumber := *number - 1
+func (api *API) GetSnapshotProposerSequence(blockNrOrHash *rpc.BlockNumberOrHash) (BlockSigners, error) {
+	var (
+		snap       *Snapshot
+		snapNumber rpc.BlockNumber
+		header     *types.Header
+		err        error
+	)
+
+	//nolint:nestif
+	if blockNrOrHash == nil {
+		header = api.chain.CurrentHeader()
```

Comment:
Do you think we should perform a `nil` header check here and at other few places below?

Reply:
Ahh I think I see what you mean : 
`snapNumber = rpc.BlockNumber(header.Number.Int64() - 1)`